egg/core/callbacks.py

```diff
@@ -44,6 +44,9 @@ def on_epoch_begin(self, epoch: int):
     def on_epoch_end(self, loss: float, logs: Interaction, epoch: int):
         pass
 
+    def on_batch(self, logs: Interaction, loss: float, data_len: int):
```

Comment:
what's data_len here, is it needed? If it's for the progress bar use case I would have this as a field of the hypothetical ProgressBarUpdater callback rather than parameter here. Especially because it's the same for every call to on_batch.
I would add a batch_id instead.
And how about changing the name to on_batch_end?

Reply:
> ProgressBarUpdater callback rather than parameter here
You're right

>I would add a batch_id instead
What would be the need for it?

> And how about changing the name to on_batch_end?
Approved

Reply:
Printing stats for the current batch or, for instance, using it to update a progressbar callback.
Same use cases of epoch_id